analyses/cell-type-ewings/scripts/singler-workflow/01-generate-tumor-ref.R

```diff
@@ -0,0 +1,118 @@
+    opt_str = c("--data_dir"),
+    type = "character",
+    default = file.path("..", "..", "data", "current", "SCPCP000015"),
+    help = "Path to folder containing all processed SCE files."
+  ), 
+  make_option(
+    opt_str = c("--auc_results_dir"),
+    type = "character",
+    default = file.path(project_root, "results", "aucell_annotation"),
+    help = "Path to folder containing output from `aucell-annotation.sh`."
+  ),
+  make_option(
+    opt_str = c("--output_reference_file"),
+    type = "character",
+    default = file.path(project_root, "scratch", "tumor-ref-singler.rds"),
+    help = "Path to file where RDS file will be saved containing all tumor cells."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Define functions -------------------------------------------------------------
+
+# subset sce to only have tumor cells present
+# if no tumor cells, then don't return anything
+subset_tumor_cells <- function(sce_file, 
+                               annotation_file){
+  
+  # pull out tumor cells from annotation file
+  tumor_cells <- readr::read_tsv(annotation_file) |> 
+    dplyr::filter(auc_classification == "Tumor") |> 
+    dplyr::pull(barcodes)
+  
+  # read in sce 
+  sce <- readr::read_rds(sce_file)
+  
+  if(length(tumor_cells) > 0){
+    
+    # filter to only contain tumor cells
+    filtered_sce <- sce[, tumor_cells]
+    
+    # pull out library id 
+    library_id <- metadata(sce)$library_id
+    
+    # add tumor labels to coldata 
+    filtered_sce$library_id <- library_id
+    filtered_sce$ref_tumor_label <- glue::glue("tumor-{library_id}")
+    
+    return(filtered_sce) 
+    
+  } else {
+    return(NULL)
+  }
+  
+}
+
+# Set up -----------------------------------------------------------------------
+
+# Define file paths to annotation files 
+annotation_file_ext <- "_auc-classifications.tsv"
+annotation_files <- list.files(opt$auc_results_dir, pattern = annotation_file_ext, recursive = TRUE, full.names = TRUE)
+
+# name file list with library ids 
+annotation_library_ids <- basename(annotation_files) |> 
+  stringr::str_remove(annotation_file_ext)
+names(annotation_files) <- annotation_library_ids
+
+
+# Define file paths to sce files 
+sce_file_ext <- "_processed.rds"
+sce_files <- list.files(opt$data_dir, pattern = sce_file_ext, recursive = TRUE, full.names = TRUE)
+
+# name file list with library ids
+sce_library_ids <- basename(sce_files) |> 
+  stringr::str_remove(sce_file_ext)
+names(sce_files) <- sce_library_ids
+
+# make sure annotation files are in the same order as sce files 
+annotation_files <- annotation_files[order(annotation_files, sce_library_ids)]
+
+# Create merged tumor cell ref -------------------------------------------------
+
+# list of tumor cell sce 
+sce_list <- purrr::map2(sce_files, annotation_files, 
+                        \(sce_file, annotation_file){
+                          subset_tumor_cells(sce_file, annotation_file)}) 
```

Comment:
Since the function takes those two arguments in that order, I don't think you need to define an inline function
```suggestion
sce_list <- purrr::map2(sce_files, annotation_files, subset_tumor_cells) 
```